src/middlewared/middlewared/plugins/smb.py

```diff
@@ -72,6 +72,125 @@ def key_cp(encoding):
             ] + initial
         }
 
+    @private
+    async def validate_admin_groups(self, sid):
+        """
+        Check if group mapping already exists because 'net groupmap addmem' will fail
+        if the mapping exists. Remove any entries that should not be present. Extra
+        entries here can pose a significant security risk. The only default entry will
+        have a RID value of "512" (Domain Admins).
+        In LDAP environments, members of S-1-5-32-544 cannot be removed without impacting
+        entire LDAP environment because this alias exists on the remote LDAP server.
+        """
+        sid_is_present = False
+        ldap = await self.middleware.call('datastore.config', 'directoryservice.ldap')
+        if ldap['ldap_enable']:
+            self.logger.debug("LDAP is enabled. Not removing extra alias entries.")
```

Comment:
This needs to explain why it is not doing that or at least rephrase it: "Extra alias entries cannot be removed while LDAP is enabled."